h2o-core/src/main/java/water/H2OListenerExtension.java

```diff
@@ -0,0 +1,16 @@
+package water;
+
+import java.util.ArrayList;
+
+public interface H2OListenerExtension {
+
+  /** Name of listener extension */
+  String getName();
+
+  /** Initialize the exception */
+  void init();
+
+  void report(String ctrl, Object... data);
+
+  ArrayList<Object[]> getData(String ctrl);
```

Comment:
this should probably not be in the API (caller should cast the object to correct implementation instead of working with interface)